src/EditorFeatures/Core/GoToDefinition/AbstractGoToSymbolService.cs

```diff
@@ -34,12 +35,12 @@ protected AbstractGoToSymbolService(IThreadingContext threadingContext, bool ass
             // We want ctrl-click GTD to be as close to regular GTD as possible.
             // This means we have to query for "third party navigation", from
             // XAML, etc. That call has to be done on the UI thread.
-            var definitions = await Task.Factory.StartNew(() =>
-                GoToDefinitionHelpers.GetDefinitions(symbol, document.Project, thirdPartyNavigationAllowed: true, cancellationToken: cancellationToken)
-                    .WhereAsArray(d => d.CanNavigateTo(document.Project.Solution.Workspace)),
-                        cancellationToken,
-                        TaskCreationOptions.None,
-                        ForegroundTaskScheduler).ConfigureAwait(false);
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.